lib/chef/resource/apt_package.rb

```diff
@@ -75,6 +84,9 @@ class AptPackage < Chef::Resource::Package
         description: "A Hash of response file variables in the form of {'VARIABLE' => 'VALUE'}.",
         default: {}, desired_state: false
 
+      property :anchor_package_name_patterns, [TrueClass, FalseClass],
```

Comment:
Would `strict` or `exact_match` be a better name?

Reply:
I picked up the name from code review suggestions https://github.com/chef/chef/pull/12945#issuecomment-1155624689. Which one do you feel is more intuitive?

Reply:
Hmm... fair point, and my suggestions risk accidental name space collision. `interpret_package_name_as_pattern` or `interpret_package_as_regex`? I'm having a hard time thinking of the polarity/orientation of `anchor_package_name_patterns` @jaymzh

Reply:
Okay, based  upon our internal conversation, we are going with `interpret_pattern_as_regex ` instead

Reply:
I don't love that, as it's a lie.

Even if you set it to false, and anchor it, it's STILL interpreted as a regex. So if I pass in "ipt.*" we'll anchor it as `^ipt.*$` which is STILL a regex. Lets not lie to our users.

Maybe `anchor_package_pattern` or something? But lets get the logic right as I mention above.

Reply:
`anchor_package_regex` then to add `^` and `$` anchors to the regex and then a true value adds and false doesn't add the anchors.

Reply:
Add `introduced: "18.3"` to this property to indicate when it was added.